File: java/problems/p0026_remove_duplicates_from_sorted_array/SolutionTest.java
package problems.p0026_remove_duplicates_from_sorted_array;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class SolutionTest {
    @Test
    void testSolution() {
        Solution solution = new Solution();

        int[] nums1 = {1, 1, 2};
        int[] expectedNums1 = {1, 2};

        int k1 = solution.removeDuplicates(nums1);

        assertEquals(k1, expectedNums1.length);
        for (int i = 0; i < k1; i++) {
            assertEquals(nums1[i], expectedNums1[i]);
        }

        int[] nums2 = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
        int[] expectedNums2 = {0, 1, 2, 3, 4};

        int k2 = solution.removeDuplicates(nums2);

        assertEquals(k2, expectedNums2.length);
        for (int i = 0; i < k2; i++) {
            assertEquals(nums2[i], expectedNums2[i]);
        }
    }
}

File: typescript/problems/p0026_remove_duplicates_from_sorted_array/solution.test.ts
import removeDuplicates from "./solution";

test("test solution", () => {
  let nums1: number[] = [1, 1, 2];
  let expectedNums1: number[] = [1, 2];

  let k1: number = removeDuplicates(nums1);

  expect(k1).toBe(expectedNums1.length);
  for (let i = 0; i < k1; i++) {
    expect(nums1[i]).toBe(expectedNums1[i]);
  }

  let nums2: number[] = [0, 0, 1, 1, 1, 2, 2, 3, 3, 4];
  let expectedNums2: number[] = [0, 1, 2, 3, 4];

  let k2: number = removeDuplicates(nums2);

  expect(k2).toBe(expectedNums2.length);
  for (let i = 0; i < k2; i++) {
    expect(nums2[i]).toBe(expectedNums2[i]);
  }
});
